14/umbraco-cms/tutorials/starter-kit/lessons/2-add-a-blog-post-publication-date/part-3.md

```diff
@@ -10,22 +10,38 @@ Finally, in Part Three we shall change the blog listing.
 
     * Select *Default.cshtml*.
 
-2. Scroll down to find the `blogpost-date` element and change it to use a nicely formatted Publication Date, i.e.:
+2. Scroll down to find the `foreach` loop.
+3. Declare a new `publicationDate` variable as the first thing with the loop:
+
+    ```csharp
+    var publicationDate = post.Value<DateTime>("publicationDate");
+    ```
+
+4. Locate the `blogpost-date` element a bit further down and change it to use the new variable:
 
     ```html
-    <small class="blogpost-date">@post.PublicationDate.ToLongDateString()</small>
+    <small class="blogpost-date">@publicationDate.ToLongDateString()</small>
     ```
 
-3. Further up in the view, you also need to redefine the `blogposts` variable:
+    * The `ToLongDateString()` method ensures a clean looking date format.
+
+5. Redefine the `blogposts` variable before the first `div` tag - this will be used for sorting the posts:
 
     ```csharp
-    var blogposts = startNode.Children.OrderByDescending(x => x.Value<DateTime>("PublicationDate")).ToList();
+    @{
+        var blogposts = Model.BlogPosts.OrderByDescending(x => x.Value<DateTime>("publicationDate")).ToList();
+    }
     ```
 
     * Because we are sorting by a custom property we need to use the generic `Value` method.
+6. Locate the `@foreach` loop, and change `Model.Blostposts` to the variable created above: `blogposts`:
+
+    ```csharp
+    @foreach (IPublishedContent post in blogposts)
+    ```
 
-4. *Save* the partial view - a confirmation message should appear confirming that the Partial view has been saved.
-5. Now view both the Blog overview and the blog posts themselves in the browser to confirm that all is working as expected.
+7. *Save* the partial view - a confirmation message should appear confirming that the Partial view has been saved.
+8. Now view both the Blog overview and the blog posts themselves in the browser to confirm that all is working as expected.
 
 ## Summary
```

Comment:
```suggestion
<details>

<summary>See the entire file: Default.cshtml</summary>

{% code title="Default.cshtml" lineNumbers="true" %}

```json
@using Umbraco.Cms.Core.Models.PublishedContent
@using Umbraco.Extensions
@model Umbraco.SampleSite.Models.LatestBlogPostsViewModel;

@{
    var blogposts = Model.BlogPosts.OrderByDescending(x => x.Value<DateTime>("publicationDate")).ToList();
}

<div class="blogposts">
    @foreach (IPublishedContent post in blogposts)
    {
        var publicationDate = post.Value<DateTime>("publicationDate");
        <a href="@post.Url()" class="blogpost">
            <div class="blogpost-meta">
                <small class="blogpost-date">@publicationDate.ToLongDateString()</small>
                <small class="blogpost-cat">
                    @await Html.PartialAsync("~/Views/Partials/CategoryLinks.cshtml", post.Value<IEnumerable<string>>("categories"))
                </small>
            </div>
            <h3 class="blogpost-title">@post.Value("pageTitle")</h3>
            <div class="blogpost-excerpt">@post.Value("excerpt")</div>
        </a>
    }

    @if (Model.BlogPosts.Count() < Model.Total)
    {
        <div class="pagination">
            <nav class="nav-bar nav-bar--center">
                @if (Model.Page <= 1)
                {
                    <span class="nav-link nav-link--black nav-link--disabled">Prev</span>
                }
                else
                {
                    <a class="nav-link nav-link--black" href="@(Model.Url + "?page=" + (Model.Page - 1))">Prev</a>
                }

                @for (int i = 1; i <= Model.PageCount; i++)
                {
                    <a class="nav-link nav-link--black @(Model.Page == i ? " nav-link--active" : null)" href="@(Model.Url + "?page=" + i)">@i</a>
                }
                @if (Model.Page == Model.PageCount)
                {
                    <span class="nav-link nav-link--black nav-link--disabled">Next</span>
                }
                else
                {
                    <a class="nav-link nav-link--black" href="@(Model.Url + "?page=" + (Model.Page + 1))">Next</a>
                }

            </nav>
        </div>
    }
</div>
```

{% endcode %}

</details>

## Summary
```

Reply:
We had received some feedback on providing the entire file code snippets at the bottom of the article for reference. Sometimes it's difficult to understand where the code should be added. Adding code snippets makes it easier to understand the code/steps.

Reply:
Ah, good point!

Looks like some formatting is messing up your suggestion here, so I'll add it locally instead 💪